### PR TITLE
Fusion MatMul Op test shouldn't use GPU for AMD/ROCm

### PR DESCRIPTION
**Pull request by rahulbatra85**
Fused MatMul Op test shouldn't use GPU for AMD/ROCm as it is not supported.

### CODE REVIEW

**Comment by gbaned** [comment]
Hi @cheshire Can you please review this PR ? Thank you!

File: tensorflow/core/kernels/matmul_op_test.cc [hunk 1]
```diff
@@ -181,6 +181,9 @@ class FusedMatMulOpTest : public OpsTestBase {
                      .Attr("transpose_b", transpose_b)
                      .Finalize(&fused_matmul));
 
+#if TENSORFLOW_USE_ROCM
+    allow_gpu_device = false;
```
Comment (cheshire) [inline tensorflow/core/kernels/matmul_op_test.cc:185]: Can we query some runtime property instead of using #if?

**Review by cheshire — CHANGES_REQUESTED** [review]

File: tensorflow/core/kernels/matmul_op_test.cc (hunk 1 above)
Comment (rahulbatra85) [reply tensorflow/core/kernels/matmul_op_test.cc]: @cheshire I have pushed new code that uses a runtime property. Please review.

Thanks!

**Review by cheshire — CHANGES_REQUESTED** [review]
Is it possible to do the check in a cleaner way? E.g. SkipTest based on platform?

**Comment by rahulbatra85** [comment]
> Is it possible to do the check in a cleaner way? E.g. SkipTest based on platform?

@cheshire I am doing it based on platform, but this is the only way I was able to get platform info. Some other fields don't differentiate between GPU platforms like CUDA or ROCm

File: tensorflow/core/kernels/matmul_op_test.cc [hunk 2]
```diff
@@ -80,6 +84,28 @@ class FusedMatMulOpTest : public OpsTestBase {
           return device.device_type() == DEVICE_GPU;
         });
 
+#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM //This needed for CPU only builds
```
Comment (cheshire) [inline tensorflow/core/kernels/matmul_op_test.cc:87]: There's no way to check GPU availability at runtime?

**Review by cheshire — CHANGES_REQUESTED** [review]

File: tensorflow/core/kernels/matmul_op_test.cc (hunk 2 above)
Comment (rahulbatra85) [reply tensorflow/core/kernels/matmul_op_test.cc:87]: I had to do this because the CPU only build would fail

**Comment by akuegel** [comment]
@ezhulenev maybe you can advice how to best check whether the test should be skipped because it is running on ROCM?
When using XLA and stream_executor, we have a way to query this information through device description, maybe there is something similar in Tensorflow?

**Comment by rahulbatra85** [comment]
Closing this for now.

**Pull request closed without merging**